shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java

```diff
@@ -0,0 +1,112 @@
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StandardPipelineDataSourceConfigurationTest {
+
+    private StandardPipelineDataSourceConfiguration standardPipelineDataSourceConfiguration;
+
+    @Before
+    public void setUp() throws SQLException {
+        standardPipelineDataSourceConfiguration = new StandardPipelineDataSourceConfiguration("jdbc:.*", "userName", "password");
+    }
+
+    @Test
+    public void assertConstructorWithStringSuccess() {
+        String parameter = "jdbcUrl: jdbc:.*\n"
+                 + "username: userName\n"
+                 + "password: password\n";
+        StandardPipelineDataSourceConfiguration standardPipelineDataSourceConfigurationOtherInstance = new StandardPipelineDataSourceConfiguration(parameter);
+        String actualDatabaseTypeName = standardPipelineDataSourceConfigurationOtherInstance.getDatabaseType().getName();
+        String expectedDatabaseTypeName = "SQL92";
+        assertThat(actualDatabaseTypeName, is(expectedDatabaseTypeName));
+    }
+
+    @Test
+    public void assertEqualsAndHashCodeSuccess() {
+        StandardPipelineDataSourceConfiguration standardPipelineDataSourceConfigurationOtherInstance = new StandardPipelineDataSourceConfiguration("jdbc:.*", "userName", "password");
+        assertThat(standardPipelineDataSourceConfiguration.hashCode(), is(standardPipelineDataSourceConfigurationOtherInstance.hashCode()));
+    }
+
+    @Test
+    public void assertGetDatabaseTypeSuccess() {
+        DatabaseType actualDatabaseType = standardPipelineDataSourceConfiguration.getDatabaseType();
+        String actualDatabaseTypeName = actualDatabaseType.getName();
+        String expectedDatabaseTypeName = "SQL92";
+        assertThat(actualDatabaseTypeName, is(expectedDatabaseTypeName));
+    }
+
+    @Test
+    public void assertGetHikariConfigSuccess() {
+        HikariConfig actualHikariConfig = standardPipelineDataSourceConfiguration.getHikariConfig();
+        long actualActualHikariConfigConnectionTimeout = actualHikariConfig.getConnectionTimeout();
+        long expectedActualHikariConfigConnectionTimeout = new Long(30000);
+        assertThat(actualActualHikariConfigConnectionTimeout, is(expectedActualHikariConfigConnectionTimeout));
+    }
+
+    @Test
+    public void assertGetParameterSuccess() {
+        String actualParameter = standardPipelineDataSourceConfiguration.getParameter();
+        String expectedParameter = "jdbcUrl: jdbc:.*\n"
+               + "username: userName\n"
+               + "password: password\n";
+        assertThat(actualParameter, is(expectedParameter));
+    }
+
+    @Test
+    public void assertGetTypeSuccess() {
+        String actualType = standardPipelineDataSourceConfiguration.getType();
+        String expectedType = "JDBC";
+        assertThat(actualType, is(expectedType));
```

Comment:
`expectedType` variable is not required, `is(expectedType)` could be `is("JDBC")`. And also other places.

I'm sorry, the code conduct doesn't fit for every case.

Reply:
Dear @sandynz, 

Thank you for your comment and collaboration.

When we should use the `expectedXXX` pattern ??

I remain always at your disposal.

Best Regards, 
Panagiotis Parthenis

Reply:
From guideline: `Pursue clean, simplified and extremely elegant codes`.
If the assign statement of `expectedXXX` is simple enough, then we could just put it into `is(...)`. e.g.
```
        assertThat(event.getType(), anyOf(is(Type.ADDED), is(Type.UPDATED)));
...
        assertThat(shardingItems.size(), is(1));
        assertThat(shardingItems.get(0), is(jobContext.getShardingItem()));
```